rust/Backtracking/combination_of_sum.rs

```diff
@@ -0,0 +1,40 @@
+fn dfs(
+    combination: &mut Vec<u32>,
+    start_index: usize,
+    nums: &[u32],
+    target: u32,
+    res: &mut Vec<Vec<u32>>,
+) {
+    // Termination condition: If the target is equal to 0, we found a combination that sums to 'k'
+    if target == 0 {
+        res.push(combination.clone());
+        return;
+    }
```

Comment:
Please add the rust equivalent below 
![image](https://github.com/user-attachments/assets/6bc91a55-2dfa-47e0-b6a6-65c0415ba38c)